esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,340 @@
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': [],
+    }
+    return record
+
+
+def rechunk_and_regrid(src, tgt, scheme):
+    """Rechunk cube src and regrid it onto the grid of cube tgt."""
+    src_chunks = compute_chunks(src, tgt)
+    src.data = src.lazy_data().rechunk(src_chunks)
+    return regrid(src, tgt, scheme)
+
+
+def change_data_type(cube):
+    """Change data type to float32."""
+    cube.data = cube.core_data().astype('float32')
+    for coord_name in 'latitude', 'longitude', 'time':
+        coord = cube.coord(coord_name)
+        coord.points = coord.core_points().astype('float32')
+        coord.bounds = None
+        coord.guess_bounds()
+    return cube
+
+
+def _convert_units(cube):
+    """Convert unit of cube.
+
+    From kg m-2 s-1 to kg m-2 month-1 or kg m-2 day-1.
+    Note that the unit kg m-2 s-1 is equivalent to mm s-1.
+    """
+    mip = cube.attributes['mip']
+
+    if mip == 'Amon':
+        cube.convert_units('kg m-2 month-1')  # equivalent to mm/month
+    elif mip == 'day':
+        cube.convert_units('kg m-2 day-1')  # equivalent to mm/day
+    return cube
+
+
+def _fix_negative_values(cube):
+    """Change negative values to zero."""
+    cube.data = da.where(cube.core_data() < 0, 0, cube.core_data())
+    return cube
+
+
+def get_input_cubes(metadata):
+    """Return a dictionary with all (preprocessed) input files."""
+    provenance = create_provenance_record()
+    all_vars = {}
+    for attributes in metadata:
+        short_name = attributes['short_name']
+        filename = attributes['filename']
+        logger.info("Loading variable %s", short_name)
+        cube = iris.load_cube(filename)
+        all_vars[short_name] = change_data_type(cube)
+        cube.attributes['mip'] = attributes['mip']
+        provenance['ancestors'].append(filename)
+    return all_vars, provenance
+
+
+def load_target(cfg):
+    """Load target grid."""
+    filename = Path(cfg['auxiliary_data_dir']) / cfg['target_file']
+    cube = iris.load_cube(str(filename))
+    return cube
+
+
+def langbein_pet(tas):
+    """Calculate potential ET using Langbein method.
+
+    Langbein curve is a temperature-based method for calculating potential ET.
+    In this equation, T is the annual average temperature in degree Celcius,
+    and pet is in mm per year.
+    Reference: https://doi.org/10.3133/cir52
+    """
+    tas.convert_units('degC')
+    constant_a = iris.coords.AuxCoord(np.float32(325),
+                                      long_name='first constant', units=None)
+    constant_b = iris.coords.AuxCoord(np.float32(21),
+                                      long_name='second constant', units=None)
+    constant_c = iris.coords.AuxCoord(np.float32(0.9),
+                                      long_name='third constant', units=None)
+
+    # assumption here: tas is constant over time, then the monthly/daily
```

Comment:
Great that you state the assumption. Also include it in the description of the scripts in the documentation.

Reply:
Assumption added to the documentation.